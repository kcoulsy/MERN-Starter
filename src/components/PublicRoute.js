import React from 'react';
import {Route, Redirect} from 'react-router-dom';

import authService from '../services/authentication';

function PublicRoute({children, ...rest}) {
    return (
        <Route
            {...rest}
            render={(props) => {
                if (authService.isLoggedIn()) {
                    return <Redirect to={{pathname: '/', state: {from: props.location}}} />;
                }

                return children;
            }}
        />
    );
}

export default PublicRoute;
