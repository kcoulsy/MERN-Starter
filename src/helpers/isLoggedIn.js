export default (token) => typeof token === 'string' && token.length > 0;
