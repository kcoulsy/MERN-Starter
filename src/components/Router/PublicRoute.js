import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import isLoggedIn from '../../helpers/isLoggedIn';

export const PublicRoute = ({ isLoggedIn, children, ...rest }) => {
    return <Route {...rest} render={publicRouteRender(isLoggedIn, children)} />;
};

export const publicRouteRender = (isLoggedIn, children) => {
    return (props) => {
        if (isLoggedIn) {
            return <Redirect to={{ pathname: '/', state: { from: props.location } }} />;
        }

        return children;
    };
};

export const mapStateToProps = (state) => {
    return {
        isLoggedIn: isLoggedIn(state.auth.token),
    };
};

export default connect(mapStateToProps)(PublicRoute);
