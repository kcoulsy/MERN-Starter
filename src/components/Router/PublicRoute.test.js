import React from 'react';
import { shallow } from 'enzyme';

import { PublicRoute, publicRouteRender, mapStateToProps } from './PublicRoute';
import { shallowToJson } from 'enzyme-to-json';
import isLoggedIn from '../../helpers/isLoggedIn';

describe('render PublicRoute component', () => {
    it('should render PublicRoute component', () => {
        const wrapper = shallow(<PublicRoute />);

        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should handle publicRouteRender when logged in', () => {
        const child = <p>JSX Child</p>;
        const isLoggedIn = true;
        const wrapper = shallow(publicRouteRender(isLoggedIn, child)({ location: jest.fn() }));

        expect(wrapper.contains(child)).toBe(false);
        expect(wrapper).toMatchSnapshot();
    });

    it('should handle publicRouteRender when not logged in', () => {
        const child = <p>JSX Child</p>;
        const isLoggedIn = false;
        const wrapper = shallow(publicRouteRender(isLoggedIn, child)({ location: jest.fn() }));

        expect(wrapper.contains(child)).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });

    it('should return correct props from store', () => {
        const testStore = {
            auth: {
                token: 'testtoken',
            },
        };
        expect(mapStateToProps(testStore)).toEqual({
            isLoggedIn: isLoggedIn(testStore.auth.token),
        });
    });
});
