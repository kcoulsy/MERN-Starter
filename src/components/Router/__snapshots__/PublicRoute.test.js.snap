// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render PublicRoute component should handle isLoggedIn prop 1`] = `
<Route
  render={[Function]}
/>
`;

exports[`render PublicRoute component should handle publicRouteRender prop 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`render PublicRoute component should handle publicRouteRender when logged in 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`render PublicRoute component should handle publicRouteRender when not logged in 1`] = `
<p>
  JSX Child
</p>
`;

exports[`render PublicRoute component should render PublicRoute component 1`] = `
<Route
  render={[Function]}
/>
`;
